examples/http/esxi_general_information.js

```diff
@@ -0,0 +1,260 @@
+  { id: 'vendor', name: 'System Vendor', path: 'hardware.systemInfo.vendor', "valueType": D.valueType.STRING },
+  { id: 'num-nics', name: 'Number of NICs', path: 'summary.hardware.numNics', "valueType": D.valueType.NUMBER },
+  { id: 'num-hbas', name: 'Number of HBAs', path: 'summary.hardware.numHBAs', "valueType": D.valueType.NUMBER },
+  { id: 'product-version', name: 'Product Version', path: 'summary.config.product.version', "valueType": D.valueType.STRING },
+  { id: 'api-version', name: 'API Version', path: 'summary.config.product.apiVersion', "valueType": D.valueType.STRING },
+  { id: 'os-type', name: 'OS Type', path: 'summary.config.product.osType', "valueType": D.valueType.STRING },
+  { id: 'product-line-id', name: 'Product Line ID', path: 'summary.config.product.productLineId', "valueType": D.valueType.STRING }
+];
+
+/**
+ * Sends a SOAP request and returns a promise with the parsed response.
+ * @param {string} body  The SOAP request body.
+ * @param {function} extractData  A call back function to extract data from body and response.
+ * @returns A promise that resolves with the parsed response.
+ */
+function sendSoapRequest (body, extractData) {
+  const d = D.q.defer()
+
+  let config ={
+    url,
+    protocol: "https",
+    rejectUnauthorized: false,
+    body,
+    jar: true
+  }
+
+  D.device.http.post(config, function (error, response, body) {
+    if (error) {
+      console.error(error)
+      D.failure(D.errorType.GENERIC_ERROR)
+    } else if (!response) {
+      D.failure(D.errorType.RESOURCE_UNAVAILABLE)
+    } else if (response.statusCode === 400) {
+      D.failure(D.errorType.AUTHENTICATION_ERROR)
+    } else if (response.statusCode !== 200) {
+      D.failure(D.errorType.GENERIC_ERROR)
+    } else {
+      const result = extractData(body)
+      d.resolve(result)
+    }
+  })
+  return d.promise
+}
+
+/**
+ * Constructs a SOAP payload with the given body content.
+ * @param {string} soapBody - The content to be included in the SOAP body.
+ * @returns {string} The complete SOAP payload as a string, including the envelope and body.
+ */
+function createSoapPayload (soapBody) {
+  return '<?xml version="1.0" encoding="UTF-8"?><soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:vim25="urn:vim25"><soapenv:Body>'
+      + soapBody +
+      '</soapenv:Body></soapenv:Envelope>'
+}
+
+/**
+ * Parses the SOAP response to extract the Session Key.
+ * @param {string} soapResponse - The SOAP response as a string.
+ * @returns {string} The Session Key extracted from the SOAP response.
+ */
+function getSessionKey(soapResponse) {
+  const $ = D.htmlParse(soapResponse)
+  return $('returnval').find('key').first().text()
+}
+
+/**
+ * Constructs and sends a SOAP login request to the ESXi server.
+ * @returns {Promise} A promise that resolves with the response of the login request.
+ */
+function login () {
```

Comment:
what if the login is unsuccessful ? There is no proper handling of that (it must call the D.failure callback with the relevant error - D.errorType.AUTHENTICATION_ERROR )